backend/controllers/mainController.js

```diff
@@ -0,0 +1,140 @@
+			//Make sure there was no error
+			if(!err) {
+				response.send(movies); //Send the result back to the client
+			} else {
+				response.sendStatus(500); //Send an error back to the client
+			}
+		});
+	});
+
+	/*
+	 * This function gets called when a client makes a GET request to "/api/movies/<something>"
+	 * It sends one specific entry from our data array to the client. 
+	 * The 'id' paramater lets us know which entry the client asked for.
+	 */
+	app.get('/api/movies/:id',function(request,response) {
+		//We use findOne() here since we just want to get one movie.
+		//Like find, the first paramater lets us narrow down our search.
+		//We're asking for the entry which has an _id equal to request.params.id.
+		Movie.findOne({_id: request.params.id},function(err,movie){
+			//Make sure there was no error
+			if(!err) {
+				response.send(movie); //Send result to client
+			} else {
+				response.sendStatus(500); //Send an error to client
+			}
+		});
+	});
+
+	/*
+	 * This function is called when the client makes a POST request to "/api/movies".
+	 * The data we need to insert is given to us in the request body. 
+	 */
+	app.post('/api/movies',function(request,response) {
+		//Collect the data we got from the client  
+		var dataToInsert = {
+			title: request.body.title,
+			description: request.body.description,
+			image: request.body.image
+		};
+		//Create a new Movie from that data
+		var newMovie = new Movie(dataToInsert);
+
+		//Actually save our new movie to the database
+		newMovie.save(function(err) {
+			//Print an error message if something went wrong
+			if(err)
+				response.send(err); //Print error if there was a problem
+			else
+				response.send(newMovie);	//Send the client the new movie we just inserted
+		});
+	});
+
+	/*
+	 * This function is called when the client makes a POST request to "/api/movies/:id/reviews".
+	 * The data we need to insert is given to us in the request body. 
+	 * We take that data and add it to the review array. 
+	 */
+	app.post('/api/movies/:id/reviews',function(request,response) {
+		
+		var movieID = request.params.id; //Movie ID will be taken from the URL   
+		//The data to use for the review will come from the request body
+		var dataToInsert = {
+			score: request.body.score,
+			body: request.body.body
+		};
+
+		//Look up the movie this review is for
+		Movie.findOne({_id: movieID},function(err,movie) {
+			//Make sure there was no error finding this movie
+			if(!err) {
+				//Add this review to the reviews arary
+				movie.reviews.push(dataToInsert);
+				//Save the review
+				movie.save(function(err) {
+					//Make sure our review was saved
+					if(err)
+						response.send(err); //Print error if there was a problem
+					else
+						response.send(movie.reviews[movie.reviews.length-1]);	//Send the client the new movie we just inserted
+				});
+			} else {
+				//We didn't find a movie with that ID. 
+				response.sendStatus(500); //Send an error to client.
```

Comment:
I also try to prevent nesting wherever I can. So I really like this paradigm.

```js
if (err) {
  return res.sendStatus(500); // return to stop execution early
}

// regular error free code execution.
```